core/src/solve.jl

```diff
@@ -256,21 +301,88 @@ function formulate!(du::AbstractVector, basin::Basin, u::AbstractVector, t::Real
         drainage = basin.drainage[i]
         infiltration = reduction_factor * basin.infiltration[i]
 
-        du[i] += precipitation - evaporation + drainage - infiltration
+        du.storage[i] += precipitation - evaporation + drainage - infiltration
     end
     return nothing
 end
 
+function get_error(pid_control::PidControl, p::Parameters)::Vector{Float64}
+    (; basin) = p
+    (; node_id, listen_node_id) = pid_control
+
+    error = zero(pid_control.derivative)
```

Comment:
Should we put this vector in the PidControl struct, so we can keep re-using the same memory rather than allocating a new vector every timestep?